File: app/src/main/java/com/kingsoft/shiyou/omnisdk/demo/java/DemoApplication.java
package com.kingsoft.shiyou.omnisdk.demo.java;

import android.content.Context;

import com.kingsoft.shiyou.omnisdk.api.OmniApplication;
import com.kingsoft.shiyou.omnisdk.api.OmniSDK;
import com.kingsoft.shiyou.omnisdk.api.entity.InitParams;
import com.kingsoft.shiyou.omnisdk.demo.common.utils.DemoLogger;

/**
 * Description: 游戏应用自身`Application`类定义。
 * <p>
 * 请游戏对接方创建定义自身应用的`Application`类，并使其继承：`com.kingsoft.shiyou.omnisdk.api.OmniApplication`。
 * 如自身有需要重写`Application`中相应的生命周期方法，并保证调用其对应的父类周期方法。
 * 同时务必在AndroidManifest.xml文件中的<application> tag标签中申明name值为该类（按自身需要加上包名前缀）。示例如下：
 * <application
 * android:name=".java.DemoApplication"
 * ...
 * >
 * ...
 * </application>
 */
public class DemoApplication extends OmniApplication {

    @Override
    public void attachBaseContext(Context base) {
        super.attachBaseContext(base);
        DemoLogger.d("DemoApplication.attachBaseContext(...) called");

        // 构建OmniSDK初始化数据配置
        InitParams initParams = new InitParams();
        // 正式版本必须设置为`false`
        initParams.setDebug(true);
        // 正式版本必须设置为`Production`
        initParams.setEnvironment("Production");
        // OmniSDK初始化数据配置接口调用
        OmniSDK.getInstance().initialize(initParams);

        // 游戏对接方自己的代码
    }

    @Override
    public void onCreate() {
        super.onCreate();
        DemoLogger.d("DemoApplication.onCreate(...) called");

        // 游戏对接方自己的代码
    }

}

File: app/src/main/java/com/kingsoft/shiyou/omnisdk/demo/kotlin/DemoApplication.kt
package com.kingsoft.shiyou.omnisdk.demo.kotlin

import android.app.Application
import android.content.Context
import androidx.multidex.MultiDex
import com.kingsoft.shiyou.omnisdk.api.OmniSDK

/**
 * Description: 若游戏应用已经有自身定义的Application并继承其他的android.app.Application子类，
 * 则需要在其自定义Application的相应方法中添加如下代码:
 *
 * @author: LuXing created on 2021/3/22 17:27
 *
 */
class DemoApplication : Application() {

    /**
     * CP对接方必须在游戏自定义的Application类的attachBaseContext生命周期方法内调用如下方法
     */
    override fun attachBaseContext(base: Context?) {
        super.attachBaseContext(base)
        // 解决方法数超过64K问题
        MultiDex.install(this)

        // CP对接方必须调用该方法
        OmniSDK.instance.onApplicationAttachBaseContext(base)

        // CP对接方自己的代码
    }

    /**
     * CP对接方必须在游戏自定义的Application类的onCreate生命周期方法内调用如下SDK API接口
     */
    override fun onCreate() {
        super.onCreate()
        // CP对接方必须调用该方法
        OmniSDK.instance.onApplicationCreate(this)

        // CP对接方自己的代码
    }

    /**
     * CP对接方必须在游戏自定义的Application类中重载onLowMemory方法并在其中调用如下SDK API接口
     */
    override fun onLowMemory() {
        super.onLowMemory()
        // CP对接方必须调用该方法
        OmniSDK.instance.onApplicationLowMemory()

        // CP对接方自己的代码
    }

    /**
     * CP对接方必须在游戏自定义的Application类中重载onTrimMemory方法并在其中调用如下SDK API接口
     */
    override fun onTrimMemory(level: Int) {
        super.onTrimMemory(level)
        // CP对接方必须调用该方法
        OmniSDK.instance.onApplicationTrimMemory()

        // CP对接方自己的代码
    }

    /**
     * CP对接方必须在游戏自定义的Application类的onTerminate生命周期方法内调用如下SDK API接口
     */
    override fun onTerminate() {
        super.onTerminate()
        // CP对接方必须调用该方法
        OmniSDK.instance.onApplicationTerminate()

        // CP对接方自己的代码
    }
}